### PR TITLE
Refactor mugshots

### PR DESCRIPTION
**Pull request by fdeblasio**
## Description
This PR applies ShinyDragonHunter's changes for allowing any trainer to have an Elite Four/Wallace-style pre-battle mugshot

## **Discord contact info**
Frankfurter0

### CODE REVIEW

File: include/data.h [hunk 1]
```diff
@@ -83,7 +85,9 @@ struct Trainer
     /*0x12*/ u8 trainerPic;
     /*0x13*/ u8 trainerName[TRAINER_NAME_LENGTH + 1];
     /*0x1E*/ bool8 doubleBattle:1;
-             u8 padding:7;
+    /*0x19*/ bool8 mugshotEnabled;
+    /*0x1A*/ u8 mugshotColor;
+    u8 padding:7;
     /*0x1F*/ u8 partySize;
```
Comment (GraionDilach) [inline include/data.h:91]:
```suggestion
             bool8 mugshotEnabled:1;
             u8 padding:6;
    /*0x1F*/ u8 partySize;
    /*0x20*/ u8 mugshotColor;
```

File: src/data.c [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #include "battle.h"
 #include "data.h"
 #include "graphics.h"
+#include "battle_transition.h"
```
Comment (AlexOn1ine) [inline src/data.c:6]: This file doesn't exist anymore.

Comment (fdeblasio) [reply src/data.c]: Removed!

File: include/data.h (hunk 1 above)
Comment (fdeblasio) [reply include/data.h]: Done!

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**